src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -307,6 +332,96 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             logger.Info(ErrorCode.SiloInitializingFinished, "-------------- Started silo {0}, ConsistentHashCode {1:X} --------------", SiloAddress.ToLongString(), SiloAddress.GetConsistentHashCode());
         }
 
+        private IServiceCollection RegisterSystemTypes()
+        {
+            //
+            // Register the system classes and grains in this method.
+            //
+
+            IServiceCollection serviceCollection = new ServiceCollection();
+
+            serviceCollection.AddTransient<ManagementGrain>();
+            serviceCollection.AddTransient<GrainBasedMembershipTable>();
+            serviceCollection.AddTransient<GrainBasedReminderTable>();
+            serviceCollection.AddTransient<PubSubRendezvousGrain>();
+            serviceCollection.AddTransient<MemoryStorageGrain>();
+
+            return serviceCollection;
+        }
+
+        private void HandleStartup(IServiceCollection serviceCollection)
+        {
+            if (String.IsNullOrWhiteSpace(nodeConfig.StartupTypeName))
+            {
+                return;
+            }
+
+            var startupType = System.Type.GetType(nodeConfig.StartupTypeName);
+
+            if (startupType == null)
+            {
+                throw new InvalidOperationException(string.Format("Can not locate the type specified in the configuration file: '{0}'.", nodeConfig.StartupTypeName));
+            }
+
+            var servicesMethod = FindConfigureServicesDelegate(startupType);
```

Comment:
I'm following the Asp.Net vNext model where the Startup class has "features". No we only have DI configuration, but - if you remember to my email - later we need to support silo startup and other startup related activities where the user can "inject" his own code without writing a custom host for Orleans.

Having an interface would make this and future methods mandatory, where they're optional.

Don't know the design decision from the part of the Asp.Net vNext team, but I never questioned this part.

Before Asp.Net vNext in Katana (the implementation of Owin) you had a Startup class without interface too.

A future advantage of this and what we can support later is a Configure method per environment. Now that part is removed from the Orleans implementation, but if you check out the Asp.Net vNext code you can have ConfigureDevServices, ConfigureAzureServices, Configure<EnvironmentName>Services named methods. This would not be possible with interfaces.